SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/Storage.cs

```diff
@@ -370,36 +357,31 @@ private Task<T> AddTaskToQueue<T>(Func<T> action)
                 catch (Exception e)
                 {
                     // And regarding the comment in other variant of this function, async lambda + await cannot work with a Func anyway.
-                    throw HandleStorageRelatedExceptionAsync(e).GetAwaiter().GetResult();
+                    throw HandleStorageRelatedExceptionAsync(e);
                 }
             });
             AddTaskToQueue(task);
             return task;
         }
 
-        private async Task<Exception> HandleStorageRelatedExceptionAsync(Exception e)
+        private Exception HandleStorageRelatedExceptionAsync(Exception e)
         {
             // Check if database is corrupted, we have evidence (https://github.com/microsoft/appcenter-sdk-dotnet/issues/1184)
             // that it's not always originated by a proper SQLiteException (which would then be converted to StorageException in StorageAdapter).
             // If it was always the right type then the exception would not have been unobserved in that application before we changed the re-throw logic here.
             // But the message is definitely "Corrupt" and thus unfortunately that is the only check we seem to be able to do as opposed to type/property checking.
             if (e.Message == "Corrupt" || e.InnerException?.Message == "Corrupt")
             {
-                AppCenterLog.Error(AppCenterLog.LogTag, "Database corruption detected, deleting the file and starting fresh...", e);
-                await _storageAdapter.DeleteDatabaseFileAsync().ConfigureAwait(false);
-                await InitializeDatabaseAsync().ConfigureAwait(false);
+                AppCenterLog.Error(AppCenterLog.LogTag,
+                    "Database corruption detected, deleting the file and starting fresh...", e);
+                _storageAdapter.Dispose();
+                File.Delete(Constants.AppCenterDatabasePath);
```

Comment:
`File.Delete` can throw `IOException`. It's not uncommon to hit the "file used by another process" error on Windows.